oauthlib/oauth2/rfc6749/endpoints/authorization.py

```diff
@@ -64,7 +64,12 @@ class AuthorizationEndpoint(BaseEndpoint):
     def __init__(self, default_response_type, default_token_type,
                  response_types):
         BaseEndpoint.__init__(self)
-        self._response_types = response_types
+        self._response_types = {}
+        # response_types are sorted internally so ordered comparison is faster/easier later
+
+        for k, v in response_types.iteritems() if sys.version_info[0] == 2 else iter(response_types.items()):
```

Comment:
I think that's probably the right decision. Nobody can really complain at us for following the letter of the spec.